website/src/pages/showcase/index.tsx

```diff
@@ -190,23 +193,53 @@ const otherUsers = sortedUsers.filter(
   (user) => !user.tags.includes('favorite'),
 );
 
+function SearchBar() {
+  const history = useHistory();
+  const location = useLocation();
+
+  return (
+    <div className={styles.searchContainer}>
+      <input
+        id="searchbar"
+        placeholder="Search for site name..."
+        defaultValue={readSearchName(location.search) ?? undefined}
+        onInput={(e) => {
+          const newSearch = new URLSearchParams(location.search);
+          newSearch.delete(SearchNameQueryKey);
+          if (e.currentTarget.value) {
+            newSearch.set(SearchNameQueryKey, e.currentTarget.value);
+          }
+          history.push({
+            ...location,
+            search: newSearch.toString(),
+            state: prepareUserState(),
+          });
+          setTimeout(() => {
+            document.getElementById('searchbar')?.focus();
+          }, 0);
```

Comment:
Not sure about this part. Because the searchbar is unmounted & remounted when switching between views (e.g. between all sites and filtered sites, or between filterd sites and "no results"), the `restoreUserState` actually doesn't correctly focus the input. I tried using `key` with no luck.

Reply:
Yes it won't work, it's not a good practice to unmount an input while typing anyway

Described in https://thoughtspile.github.io/2022/01/17/jsx-conditionals/, `# Remount or update?`

It would require some additional refactorings to make it work better

Reply:
Yes, we probably need to refactor the DOM structure and keep a persisted container. Will look at it later.